src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs

```diff
@@ -47,6 +47,15 @@ public static bool IsLambda(SyntaxNode node)
         public static SyntaxNode GetLambda(SyntaxNode lambdaBody)
         {
             var lambda = lambdaBody.Parent;
+            if (lambda.Kind() == SyntaxKind.ArrowExpressionClause)
```

Comment:
We should consider the `ArrowExpressionClause` to be the body of the expression-bodied local function, so this is unnecessary. Instead, we should assert that the user isn't passing an expression as the body of an expression-bodied local function.